### PR TITLE
feat(dlp): added a sample for dlp_deidentify_table_with_crypto_hash

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description
- Added a sample for `dlp_deidentify_table_with_crypto_hash` with a unit test case.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by telpirion — APPROVED** [review]

File: dlp/snippets/deid/deid_test.go [hunk 1]
```diff
@@ -570,3 +570,28 @@ func TestReidTextDataWithFPE(t *testing.T) {
 		t.Errorf("reidentifyFreeTextWithFPEUsingSurrogate got %q, want %q", got, want)
 	}
 }
+
+func TestDeIdentifyTableWithCryptoHash(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	var buf bytes.Buffer
+	transientKey := "YOUR_TRANSIENT_CRYPTO_KEY"
```
Comment (annadata) [inline dlp/snippets/deid/deid_test.go:578]: Please use transientKeyName both variable name(s) and text.

File: dlp/snippets/deid/deid_test.go [hunk 2]
```diff
@@ -570,3 +570,28 @@ func TestReidTextDataWithFPE(t *testing.T) {
...
+	if err := deIdentifyTableWithCryptoHash(&buf, tc.ProjectID, transientKey); err != nil {
+		t.Fatal(err)
+	}
+	got := buf.String()
+
+	if want := "Table after de-identification :"; !strings.Contains(got, want) {
+		t.Errorf("TestDeIdentifyTableWithCryptoHash got %q, want %q", got, want)
+	}
+	if want := `{string_value:\"my email is user3@example.org and phone is 858-555-0224\"}`; strings.Contains(got, want) {
+		t.Errorf("TestDeIdentifyTableWithCryptoHash got %q, want %q", got, want)
```
Comment (annadata) [inline dlp/snippets/deid/deid_test.go:589]: Can you please split this to confirming email address is not present and phone number is not present.

**Review by annadata — APPROVED** [review]

File: dlp/snippets/deid/deid_test.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Updated.

File: dlp/snippets/deid/deid_test.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: updated.

**Pull request merged**